pkg/controller/scylladbmanagertask/sync_manager.go

```diff
@@ -0,0 +1,608 @@
+		ObservedGeneration: smt.Generation,
+		Reason:             "UpdatedScyllaDBManagerTask",
+		Message:            fmt.Sprintf("Updated a ScyllaDB Manager task: %s (%s).", managerTask.Name, managerTaskID),
+	})
+
+	return progressingConditions, nil
+}
+
+func getScyllaDBManagerClientTask(ctx context.Context, smt *scyllav1alpha1.ScyllaDBManagerTask, clusterID string, managerClient *managerclient.Client) (*managerclient.TaskListItem, bool, error) {
+	taskName := scyllaDBManagerClientTaskName(smt)
+
+	taskType, err := scyllaDBManagerClientTaskType(smt)
+	if err != nil {
+		return nil, false, fmt.Errorf("can't get ScyllaDB Manager client task type: %w", err)
+	}
+
+	var taskID string
+	if smt.Status.TaskID != nil {
+		taskID = *smt.Status.TaskID
+	}
+
+	tasks, err := managerClient.ListTasks(ctx, clusterID, taskType, true, "", taskID)
+	if err != nil {
+		klog.V(4).InfoS("Failed to list ScyllaDB Manager client tasks.", "ScyllaDBManagerTask", klog.KObj(smt), "ScyllaDBManagerClientClusterID", clusterID, "ScyllaDBManagerClientTaskType", taskType, "ScyllaDBManagerClientTaskID", taskID, "Error", err)
+		return nil, false, fmt.Errorf("can't list ScyllaDB Manager client tasks: %s", managerclienterrors.GetPayloadMessage(err))
+	}
+
+	if len(tasks.TaskListItemSlice) == 0 {
+		return nil, false, nil
+	}
+
+	if len(taskID) > 0 && len(tasks.TaskListItemSlice) > 1 {
+		return nil, false, fmt.Errorf("more than one task found in ScyllaDB Manager state with taskID: %s", taskID)
+	}
+
+	idx := slices.IndexFunc(tasks.TaskListItemSlice, func(item *managerclient.TaskListItem) bool {
+		return item.Name == taskName
+	})
+
+	if idx >= 0 {
+		return tasks.TaskListItemSlice[idx], true, nil
+	}
+
+	return nil, false, nil
+}
+
+type ScyllaDBManagerClientTaskOverrideOption func(*scyllav1alpha1.ScyllaDBManagerTask, *managerclient.Task)
+
+// WithScheduleStartDateNowSyntaxRetention ensures backward compatibility for tasks created with the deprecated "now" syntax in their start date override annotation.
+// On an update, tasks with an already set start date retain it, rather than have a new, recalculated date time set, to prevent the schedule from being restarted continuously.
+func WithScheduleStartDateNowSyntaxRetention(existingStartDate strfmt.DateTime) func(*scyllav1alpha1.ScyllaDBManagerTask, *managerclient.Task) {
+	return func(smt *scyllav1alpha1.ScyllaDBManagerTask, managerTask *managerclient.Task) {
+		startDateOverrideAnnotation := smt.Annotations[naming.ScyllaDBManagerTaskScheduleStartDateOverrideAnnotation]
+		if !strings.HasPrefix(startDateOverrideAnnotation, "now") {
+			return
+		}
+
+		if managerTask.Schedule == nil {
+			managerTask.Schedule = &managerclient.Schedule{}
+		}
+
+		managerTask.Schedule.StartDate = pointer.Ptr(existingStartDate)
+	}
+}
+
+func makeScyllaDBManagerClientTask(smt *scyllav1alpha1.ScyllaDBManagerTask, clusterID string, overrideOptions ...ScyllaDBManagerClientTaskOverrideOption) (*managerclient.Task, error) {
```

Comment:
Thought: `makeScyllaDBManagerClientTask` and all its dependencies would be a good candidate for a separate PR. We could implement it in a separate, helper package, test it there, just not plug it into the controller yet. (no action required)